src/factory/__tests__/mongoDBService.test.ts

```diff
@@ -0,0 +1,108 @@
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+import * as TE from "fp-ts/TaskEither";
+import { left, right } from "fp-ts/lib/Either";
+import { Collection, Db, MongoClient } from "mongodb";
+import { ContinuationTokenItem } from "../../capturer/cosmos/utils";
+import * as MongoUtils from "../../capturer/mongo/utils";
+import { mongoDBService } from "../mongoDBService";
+
+const mockMongoClient: MongoClient = {} as MongoClient;
+const mockDatabase: Db = {} as Db;
+const mockCollection: Collection = {} as Collection;
+const mockItem: ContinuationTokenItem = { id: "test" } as ContinuationTokenItem;
+
+jest.mock("../../capturer/mongo/utils");
+
+const mongoConnectSpy = jest.spyOn(MongoUtils, "mongoConnect");
+const getMongoDbSpy = jest.spyOn(MongoUtils, "getMongoDb");
+const getMongoCollectionSpy = jest.spyOn(MongoUtils, "getMongoCollection");
+const findDocumentByIDSpy = jest.spyOn(MongoUtils, "findDocumentByID");
+
+describe("mongoDBService", () => {
+  beforeEach(() => {
+    jest.resetModules();
+    jest.clearAllMocks();
+  });
+
+  it("should connect to Mongo successfully", async () => {
+    mongoConnectSpy.mockImplementationOnce(() => TE.right(mockMongoClient));
+    const result = await mongoDBService.connect({
+      connection: "valid-connection",
+    })();
+    expect(E.isRight(result)).toBeTruthy();
+    expect(result).toEqual(right(mockMongoClient));
+  });
+
+  it("should handle connection error", async () => {
+    mongoConnectSpy.mockImplementationOnce(() =>
+      TE.left(new Error("Connection error"))
+    );
+    const result = await mongoDBService.connect({
+      connection: "invalid-connection",
+    })();
+    expect(E.isLeft(result)).toBeTruthy();
+    expect(result).toEqual(left(new Error("Connection error")));
+  });
+
+  it("should get database successfully", () => {
+    getMongoDbSpy.mockImplementationOnce(() => right(mockDatabase));
+    const result = mongoDBService.getDatabase(mockMongoClient, "test-database");
+    expect(E.isRight(result)).toBeTruthy();
+    expect(result).toEqual(right(mockDatabase));
+  });
+
+  it("should handle error when getting database", () => {
+    getMongoDbSpy.mockImplementationOnce(() =>
+      left(new Error("Database error"))
+    );
+    const result = mongoDBService.getDatabase(
+      mockMongoClient,
+      "invalid-database"
+    );
+    expect(E.isLeft(result)).toBeTruthy();
+    expect(result).toEqual(left(new Error("Database error")));
+  });
+
+  it("should get resource successfully", async () => {
+    getMongoCollectionSpy.mockImplementationOnce(() =>
+      TE.rightTask(() => Promise.resolve(mockCollection))
```

Comment:
just as a suggestion we can safely change this expression with another one like:
```suggestion
      TE.right(mockCollection)()
```
or even
```suggestion
     T.of(mockCollection)
```

with `T` as `import * as T from "fp-ts/lib/Task"`